rules/Transform/Rector/Class_/AddAllowDynamicPropertiesAttributeRector.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+namespace Rector\Transform\Rector\Class_;
+
+use PhpParser\Node;
+use PhpParser\Node\Stmt\Class_;
+use Rector\Core\Rector\AbstractRector;
+use Rector\Php80\NodeAnalyzer\PhpAttributeAnalyzer;
+use Rector\PhpAttribute\Printer\PhpAttributeGroupFactory;
+use Symplify\RuleDocGenerator\ValueObject\CodeSample\CodeSample;
+use Symplify\RuleDocGenerator\ValueObject\RuleDefinition;
+
+/**
+ * @changelog https://wiki.php.net/rfc/deprecate_dynamic_properties
+ *
+ * @see \Rector\Tests\Transform\Rector\Class_\AddAllowDynamicPropertiesAttributeRector\AddAllowDynamicPropertiesAttributeRectorTest
+ */
+class AddAllowDynamicPropertiesAttributeRector extends AbstractRector
```

Comment:
I think `implements MinPhpVersionInterface` still needed so it indicate that it will only run in a version allowed, eg:

```php
PhpVersionFeature::DEPRECATE_DYNAMIC_PROPERTIES
```